client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -145,7 +150,11 @@ export default function Confirm( props: WooCommerceInstallProps ): ReactElement
 		);
 	}
 
-	if ( ! siteId || ! isDataReady || isAtomicSite || isReadyForTransfer ) {
+	if (
+		! siteId ||
+		! isDataReady ||
+		( ( isAtomicSite || isReadyForTransfer ) && ! siteUpgrading?.required )
```

Comment:
I think this is fine to get this fixed, but we should come back to this and refactor it, as the logic has `isReadyForTransfer` as true, and it's not. 

I think we're using that on the landing page, instead we should look for eligibility holds.